llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -166,6 +166,42 @@ static bool canUseShiftPair(Instruction *Inst, const APInt &Imm) {
   return false;
 }
 
+// If this is a 64-bit AND with a mask of the form -(1 << C) in the lower 32
+// bits and the only user is an equality comparison, we might be able to use a
+// sraiw instead. This avoids the need to materialize the AND constant.
+static bool canUseSRAIWCmp(Instruction *Inst, const APInt &Imm) {
+  if (!Inst->hasOneUse())
```

Comment:
Multi-use doesn't seem much harder: just add a `for` loop and only `return true` after all users match.

Reply:
I don't think the DAGCombine supports multiple use.

Reply:
Ok, I didn't know that.